src/api/v2/compatibility/application/CompatibleApplicationResourceTreeItem.ts

```diff
@@ -0,0 +1,108 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzExtParentTreeItem, AzExtResourceType, AzExtTreeDataProvider, AzExtTreeItem, IActionContext, ISubscriptionContext, nonNullProp, TreeItemIconPath } from "@microsoft/vscode-azext-utils";
+import type { ResolvedAppResourceBase } from "@microsoft/vscode-azext-utils/hostapi";
+import { TreeItemCollapsibleState } from "vscode";
+import { createResolvableProxy } from "../../../../tree/AppResourceTreeItem";
+import { getIconPath } from "../../../../utils/azureUtils";
+import { AzureResource } from "../../v2AzureResourcesApi";
+
+/**
+ * Must immitate the behavior of AppResourceTreeItem
+ */
+export class CompatibleResolvedApplicationResourceTreeItem extends AzExtParentTreeItem {
```

Comment:
If the intent of this class is to imitate the behavior of the `AppResourceTreeItem`, should this also implement `GroupableResource, AppResource` like `AppResourceTreeItem` does? 

In doing so, it reveals that `groupConfig` isn't defined. Though I wouldn't think that should be necessary to have since the original `AppResourceTreeItem` before it is resolved contains that.

Following that line of thinking, isn't this imitating the `ResolvedAppResourceBase` more than an `AppResourceTreeItem`?

Reply:
I'm not sure I follow exactly, but

> Following that line of thinking, isn't this imitating the ResolvedAppResourceBase more than an AppResourceTreeItem?

That is exactly correct, this is imitating a `ResolvedAppResourceBase`. I will update the comment.

Reply:
Comment updated